lib/vscode_helper.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.8
+
+import 'dart:convert';
+
+import 'package:file/file.dart';
+import 'package:flutter_tools/src/convert.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/project.dart';
+
+const String kConfigNameAttach = 'flutter-tizen: Attach';
+
+final RegExp _commentFormat =
+    RegExp(r'(:?)(/\*([^*]|[\r\n]|(\*+([^*/]|[\r\n])))*\*+/|//.*\n?)');
+
+String _removeComments(String jsonString) {
+  return jsonString.replaceAllMapped(
+    _commentFormat,
+    (Match match) {
+      // "aaa://bbb" contains "//" but is not a comment.
+      return match.group(1) == ':' ? match.group(0) : '';
+    },
+  ).trim();
+}
+
+void updateLaunchJsonFile(FlutterProject project, Uri observatoryUri) {
+  if (project.directory.basename == 'example') {
+    final FlutterProject parentProject =
+        FlutterProject.fromDirectory(project.directory.parent);
+    if (parentProject.isPlugin || parentProject.isModule) {
+      updateLaunchJsonFile(parentProject, observatoryUri);
+    }
+  }
+
+  final Directory vscodeDir = project.directory.childDirectory('.vscode')
+    ..createSync(recursive: true);
+  final File launchJsonFile = vscodeDir.childFile('launch.json');
+  String jsonString = '';
+  if (launchJsonFile.existsSync()) {
+    // Comments are allowed in launch.json file. Remove them to prevent
+    // decoding errors.
+    jsonString = _removeComments(launchJsonFile.readAsStringSync());
+  }
+
+  Map<dynamic, dynamic> decoded = <dynamic, dynamic>{};
+  if (jsonString.isNotEmpty) {
+    try {
+      decoded = jsonDecode(jsonString) as Map<dynamic, dynamic>;
+    } on FormatException catch (error) {
+      globals.printError('Failed to parse ${launchJsonFile.path}:\n$error');
+      return;
```

Comment:
If failed,why not just create a new one?

Reply:
The file is not owned by the flutter-tizen tool but is managed by VS Code and the user, so we don't want to overwrite it arbitrarily.

Side note)
The main reason for the FormatException should be manually edited `launch.json` files which contain invalid "trailing commas". Similarly to "comments", trailing commas are not allowed in standard JSON but allowed in `launch.json` files which use extended JSON. I need to find a better way to handle this.